File: src/백도현/week6/6-2/boj15649_dh.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split());
visited = [0] * (N + 1)
global arr
arr = []

def dfs(i):
    arr.append(i)
    visited[i] = 1
    if len(arr) == M:
        print(' '.join(map(str, arr)))
    else:
        for j in range(1, N + 1):
            if visited[j] == 0:
                dfs(j)
    visited[i] = 0
    arr.pop()

for i in range(1, N + 1):
    dfs(i)

File: src/백도현/week6/6-2/boj15649_dh.java
import java.io.*;
import java.util.*;

public class Main {
    static int N, M;
    static boolean[] visited;
    static int[] arr;


    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

		N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());

        arr = new int[M];
        visited = new boolean[N];


        dfs(N, M, 0);
        br.close();
    }

    public static void dfs(int N, int M, int depth) {
        if (depth == M) {
            for(int i = 0; i < M-1; i++) {
                System.out.print(arr[i] + " ");
            }
            System.out.print(arr[M-1]);
            System.out.println();

            return;
        }

        for (int i = 0; i < N; i++) {
            if (!visited[i]) {
                visited[i] = true;
                arr[depth] = i + 1;
                dfs(N, M, depth + 1);
                visited[i] = false;
            }
        }
        return ;
    }

}